src/Executables/Game/Packets/Shop/ShopOpen.cs

```diff
@@ -11,6 +11,48 @@ public partial class ShopOpen
         public ushort Size => (ushort)Items.Length;
         public uint Vid { get; set; }
 
-        public ShopItem[] Items { get; set; } = new ShopItem[40];
+        public ShopItem[] Items { get; set; } = new ShopItem[40]
+        {
+            new ShopItem(),
```

Comment:
Instead of initializing 40 empty objects, can't you just create an array list and use it instead?

Reply:
Do u mean https://learn.microsoft.com/en-us/dotnet/api/system.collections.arraylist ?

Because that type is kinda outdated